packages/core/database/src/entity-manager/index.ts

```diff
@@ -834,16 +834,63 @@ export const createEntityManager = (db: Database): EntityManager => {
 
             const { idColumn, typeColumn } = morphColumn;
 
-            await this.createQueryBuilder(joinTable.name)
-              .delete()
-              .where({
-                [idColumn.name]: id,
-                [typeColumn.name]: uid,
-                ...(joinTable.on || {}),
-                field: attributeName,
-              })
-              .transacting(trx)
-              .execute();
+            // for connect/disconnect, only modify those relations
```

Comment:
I think you are missing the `set` usecase too

Reply:
For `set` they all need to be deleted and then recreated. I could actually check if `set` exists, and then skip this entire step, because I would consider that a `set` overrides any connect/disconnect that is used (and ideally we would throw an error if you tried to combine them, I don't know if that's the case or not)

Reply:
yes if set is defined the connect & disconnect shouldn't be. 

Right now the code will just use the data to set but not access the data inside the set key so it would break